src/Analyzers/CSharp/Tests/AddParameter/AddParameterTests.cs

```diff
@@ -3158,4 +3153,148 @@ public void Main()
             }
             """);
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71428")]
+    public async Task TestAddConstructorParameterWithExistingField_BlockInitialize()
+    {
+        await TestInRegularAndScript1Async(
+            """
+            class C
+            {
+                private readonly string s;
+                private readonly string t;
+                private readonly int i;
+
+                public C(string s, int i)
+                {
+                    this.s = s;
+                    this.i = i;
+                }
+            }
+
+            class D
+            {
+                void M(string t)
+                {
+                    new [|C|]("", t, 0);
+                }
+            }
+            """,
+            """
+            class C
+            {
+                private readonly string s;
+                private readonly string t;
+                private readonly int i;
+
+                public C(string s, string t, int i)
+                {
+                    this.s = s;
+                    this.t = t;
+                    this.i = i;
+                }
+            }
+            
+            class D
+            {
+                void M(string t)
+                {
+                    new C("", t, 0);
+                }
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71428")]
+    public async Task TestAddConstructorParameterWithExistingField_ExpressionBodyInitialize()
+    {
+        await TestInRegularAndScript1Async(
+            """
+            class C
+            {
+                private readonly string s;
+                private readonly string t;
+
+                public C(string s)
+                    => this.s = s;
+            }
+
+            class D
+            {
+                void M(string t)
+                {
+                    new [|C|]("", t);
+                }
+            }
+            """,
+            """
+            class C
+            {
+                private readonly string s;
+                private readonly string t;
+
+                public C(string s, string t)
+                {
+                    this.s = s;
+                    this.t = t;
+                }
```

Comment:
here we get the logic for free that the expression body is rewritten into a block body, with the assignment correctly placed.